assets/content/cookbook/Graphics/Creating-SVG/01.Drawing-a-Circle.md

```diff
@@ -0,0 +1,313 @@
+# SVG - Drawing a Circle
+
+Dynamically creating SVG within the browser with JavaScript can be painful. 
+With Haxe JavaScript we can use abstracts to hide the uglier parts of working with SVG.
+In this tutorial we will use abstracts to help make manipulation of SVG simpler. 
+It is assumed you have explored abstracts in Haxe in depth before trying this tutorial and have a rough understanding of SVG.
+
+## Summary First
+With Haxe's abstracts it's possible to create SVG with really clean code and the structural overheads largely disappear in the compiled JavaScript. 
+
+Here is an example of the Main class we will create:
+
+```haxe
+import svgShapes.SvgRoot;
+import svgShapes.SvgCircle;
+
+class MainCircle {
+  public static function main() {
+    new MainCircle();
+  }
+  
+  var svgRoot:SvgRoot;
+    
+  public function new() {
+    svgRoot = new SvgRoot();
+    svgRoot.width = 800;
+    svgRoot.height = 800;
+    createCircle();
+  }
+
+  public function createCircle(){
+    var svgCircle = new SvgCircle();
+    svgCircle.cx = 200;
+    svgCircle.cy = 200;
+    svgCircle.r = 50;
+    svgCircle.setFillAndAlpha(0xff0000, 1);
+    svgRoot.add(svgCircle);
+  }
+}
+```
+Lets look at how SVG is created.
+
+## Creating SVG node and element
+
+To create the nodes of SVG we create SVGElements using a string type and then set attributes. 
+So a SVG circle node would be created:
+
+```haxe
+var nameSpace = "http://www.w3.org/2000/svg"
+var svgElement:SVGElement = cast Browser.document.createElementNS(nameSpace, 'circle');
+```
+Its attributes can be set, such as the radius:
+```haxe
+svgElement.setAttribute( 'r','50' );
+```
+
+We can also get its attributes:
+
+```haxe
+var radius = svgELement.getAttribute('r');
+```
+
+## Setting up SVG
+
+To create SVG we create a root node which we append into our HTML document.
+```haxe 
+// create SVGElement
+var svgElement: SVGElement = cast Browser.document.createElementNS("http://www.w3.org/2000/svg", 'svg');
+// Append to the document body
+Browser.document.body.appendChild(svgElement);
+```
+With SVG or canvas we need to setup the style so that coordinates are relative to the top corner.
+This is done using the 'style' property of the svg node.
+```haxe
+var style:CSSStyleDeclaration = element.style;
+style.paddingLeft = "0px";
+style.paddingTop = "0px";
+style.left = "0px";
+style.top = "0px";
+style.position = "absolute";
+```
+It is important to set the width and height attributes, those can be set like this:
+```haxe
+svgElement.setAttribute('width', '800');
```

Comment:
Hey, why use setAttribute here not set just [`svgElement.width`](http://api.haxe.org/js/html/svg/SVGElement.html#width)?